omegalambda/main/observing/condition_checker.py

```diff
@@ -317,23 +326,34 @@ def cloud_check(self):
 
         """
         satellite = self.config_dict.cloud_satellite
-        day = str(int(time_utils.days_of_year())).zfill(3)
+        day = int(time_utils.days_of_year())
         conus_band = 13
-        _time = datetime.datetime.now(datetime.timezone.utc)
+        _time = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(minutes=5)
         year = _time.year
         time_round = time_utils.rounddown_300(_time.hour * 60 * 60 + _time.minute * 60 + _time.second)
         req = None
         s = requests.Session()
         for i in range(6):
             hour = int(time_round / (60 * 60))
             minute = int((time_round - hour * 60 * 60) / 60) - i
+            if minute < 0:
+                hour -= 1
+                minute += 60
+            if hour < 0:
+                day -= 1
+                hour += 24
+            if day < 0:
+                year -= 1
+                day += 365
             _time = '{0:02d}{1:02d}'.format(hour, minute)
             if (minute - 1) % 5 != 0:
                 continue
+            daystr = str(day).zfill(3)
             url = 'https://www.ssec.wisc.edu/data/geo/images/goes-16/animation_images/' + \
```

Comment:
This should be a config value or at the very least a global variable at the top of the file.  URLs change often and unexpectedly and you will want to be able to easily reflect that.